### PR TITLE
Fix `random.random()`

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lcompilers/lpython/issues/2382

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
Related PR https://github.com/lfortran/lfortran/pull/3156

A Test is added in the related PR.

Also manual test:
```bash
% cat examples/expr2.py 
from lpython import f64
import random

def test_seed():
    random.seed()
    t1: f64 = random.random()
    print(t1)

test_seed()
% lpython examples/expr2.py
1.80525214961043207e-01
% lpython examples/expr2.py
8.56492263663789360e-02
% lpython examples/expr2.py
6.91035861936880158e-01
% lpython examples/expr2.py
6.65666734644056679e-01
% lpython examples/expr2.py
6.16492530618092283e-01
% lpython examples/expr2.py
2.97986548067064261e-01
% lpython examples/expr2.py
4.91210944713657227e-01
% lpython examples/expr2.py
8.56492263663789360e-02
% lpython examples/expr2.py
3.64365308249539388e-01
% lpython examples/expr2.py
3.07370851425161074e-01
% lpython examples/expr2.py
3.32739978717984608e-01
% lpython examples/expr2.py
4.64277766861150853e-01
% lpython examples/expr2.py
5.95815555004317154e-01
% lpython examples/expr2.py
4.84954742474926037e-01
% lpython examples/expr2.py
3.07370851425161074e-01
% lpython examples/expr2.py
1.26658859256030459e-01
% lpython examples/expr2.py
6.79743189681201843e-01
% lpython examples/expr2.py
4.67405867980516476e-01
% lpython examples/expr2.py
7.43338167082210122e-01
% lpython examples/expr2.py
8.85824309143155908e-01
% lpython examples/expr2.py
4.20140032852133749e-01
% lpython examples/expr2.py
3.44032650973663034e-01
% lpython examples/expr2.py
9.11193436435979498e-01
% lpython examples/expr2.py
1.77397113841677612e-01
% lpython examples/expr2.py
2.67925269095192320e-01
% lpython examples/expr2.py
6.11800378939043932e-01
% lpython examples/expr2.py
9.28742310930389170e-01
% lpython examples/expr2.py
5.38821098179938729e-01
% lpython examples/expr2.py
6.38733556791550305e-01
% lpython examples/expr2.py
9.69751943820040596e-01
% lpython examples/expr2.py
1.89261478460049960e-02
% lpython examples/expr2.py
7.01984215854659754e-01

```

**Comment by Shaikh-Ubaid** [comment]
The CI `macos` test runs indefinitely. This is also experienced in the `main` branch CI run.

**Review by certik — APPROVED** [review]

File: integration_tests/test_random.py [hunk 1]
```diff
@@ -72,12 +72,14 @@ def test_seed():
...
     t7: f64 = random.random()
+
+    print(t1, t2, t3, t4, t5, t6, t7)
     assert t1 != t2
     assert t1 == t3
     assert t1 != t4
     assert t1 != t5
     assert t4 == t5
-    assert t6 != t7
+    # assert t6 != t7
```
Comment (Shaikh-Ubaid) [inline integration_tests/test_random.py:82]: @certik  I commented this out as we discussed in the meet.

**Review by certik — APPROVED** [review]
That should be good.

**Comment by Shaikh-Ubaid** [comment]
The `macos` CI failed thrice in this PR. It seems like it is more of an actual issue than a flaky one. Since the `macos` CI also fails in the `main` branch, I think it is fine to merge this PR. @certik I could not find any option to merge a PR when the CI does not pass completely. Do you know how to merge this PR?

**Comment by certik** [comment]
Yes, the CI issue is real, something goes wrong.

**Pull request merged**